### PR TITLE
Add --disable-audio-cache startup parameter

### PR DESCRIPTION
**Pull request by michaelherger**
Disable caching of downloaded audio files at runtime. Comes in handy when running librespot on a small device with SD card or other small storage.

### CODE REVIEW

**Comment by awiouy** [comment]
Thanks! I am using this now, and I hope it will make it upstream.

**Comment by sashahilton00** [comment]
Using this now as well, +1 for upstream :)

**Comment by plietar** [comment]
Thanks @michaelherger 

Could you move the logic to the `Cache::save_file` function ?

**Comment by mherger** [comment]
I put it in the player code as I don't want to disable all (potential) file caching. It's really about caching the audio data. There could be other code using the cache. (FWICT there's none right now, is there?)

**Comment by plietar** [comment]
Yes `save_file` is only for audio data, so it is fine to move it there

**Comment by michaelherger** [comment]
Ok, will update my change.

(argh... at some point I have to have a better strategy to separate my pro from my personal GitHub account...)

**Comment by michaelherger** [comment]
Hmm... credentials is using the same cache object all over the place. Not initializing cache for files only, but use it for credentials looks like a much more complicated change, dealing with multiple cache instances etc. Am I totally wrong?

**Comment by michaelherger** [comment]
BTW: is cache/default_cache even used anywhere?

**Comment by plietar** [comment]
My point was to make `use_audio_cache` a field of `Cache` rather than `Config`, and move the `if use_audio_cache` test into the `save_file` method.

`default_cache` is just some leftover from a previous implementation, I forgot to delete it. You can do it if you want

**Comment by michaelherger** [comment]
Take 2.

File: src/main.rs [hunk 1]
```diff
@@ -135,7 +136,7 @@ fn setup(args: &[String]) -> Setup {
     let device_id = librespot::session::device_id(&name);
 
     let cache = matches.opt_str("c").map(|cache_location| {
-        Cache::new(PathBuf::from(cache_location))
+        Cache::new(PathBuf::from(cache_location), !matches.opt_present("disable-audio-cache"))
```
Comment (plietar) [inline src/main.rs]: Just a nit, could you extract `!matches.opt_present("disable-audio-cache")` to a separate variable ?

**Review by plietar — CHANGES_REQUESTED** [review]

**Comment by plietar** [comment]
Awesome thanks !

**Pull request merged**